src/custom-elements/src/components/badge/badge.tsx

```diff
@@ -0,0 +1,91 @@
+import { Component, Element, Prop, State, Watch } from '@stencil/core';
+
+@Component({
+  tag: 'chi-badge',
+  styleUrl: 'badge.scss',
+  scoped: true
+})
+export class Badge {
+  @State() slotBadgeContent = true;
+
+  @Element() el: HTMLElement;
+
+  /**
+   *  to set mode of a badges { outline, flat }.
+   */
+  @Prop({ reflectToAttr: true }) mode: string;
```

Comment:
I have just realized reviewing the results that we cannot use mode as an attribute name. Stencil changes scope of the element by adding the mode to the scope name. It is not documented. I removed chi.css from the document and this is how it is seen: 
<img width="1290" alt="Screen Shot 2019-06-21 at 12 16 56 PM" src="https://user-images.githubusercontent.com/43439749/59916197-cb2a7d80-941e-11e9-8c99-f82f1711e0ac.png">

It should look like this: 
<img width="1308" alt="Screen Shot 2019-06-21 at 12 15 42 PM" src="https://user-images.githubusercontent.com/43439749/59916210-d5e51280-941e-11e9-981c-c461d83543e9.png">

The reason is because stencil changes the scope name: 
<img width="826" alt="Screen Shot 2019-06-21 at 12 20 14 PM" src="https://user-images.githubusercontent.com/43439749/59916325-20668f00-941f-11e9-8257-2ef2073b6647.png">

Reply:
I have changed the attribute name from mode -> badge-mode and have updated the PR. Thanks.
I see one issue with capitalized is that it just sets the first letter to uppercase but doesn't set the rest of the uppercase letters to lowercase. Any thoughts @jllr @dani-cl-madrid